doc/requirements.txt

```diff
@@ -1,3 +1,3 @@
-sphinx==4.3.0
+sphinx==5.0.0
```

Comment:
I think this should be out of scope for this PR, maybe you can elaborate?

Reply:
<img width="662" alt="Screenshot 2024-01-15 at 2 27 37 PM" src="https://github.com/gitpython-developers/GitPython/assets/142979605/a41285f5-70aa-44a2-a148-5e463ad0cd5a">

I'm trying to fix this CI/CD error , but it seems useless (, anyway I'll remove the commits about this

Reply:
I have proposed a fix in #1803 for the Sphinx bug discovered here, which as noted is not specific to or caused by any of the changes in this pull request (see also #1802 for details on it). If #1803 is merged to main, then this pull request could be updated by rebasing it onto main to eliminate the issue here too.